packages/@aws-cdk/aws-appmesh/lib/gateway-route-spec.ts

```diff
@@ -143,28 +360,64 @@ class HttpGatewayRouteSpec extends GatewayRouteSpec {
    */
   readonly routeType: Protocol;
 
+  /**
+   * The default target host name to write to.
+   * Boolean is used to select either enable or disable.
+   * When enabled, rewrites the original request received at the Virtual Gateway to the destination Virtual Service hostname.
+   * When disabled, does not rewrite to destination Virtual Service hostname.
+   *
+   * @default true
+   */
+  readonly defaultHostnameRewrite?: boolean;
+
+  /**
+   * The path to rewrite.
+   *
+   * @default - rewritten to the target Virtual Service's hostname and the matched prefix is rewritten to '/'.
+   */
+  readonly pathRewrite?: HttpGatewayRouteRewrite;
+
   constructor(options: HttpGatewayRouteSpecOptions, protocol: Protocol.HTTP | Protocol.HTTP2) {
     super();
     this.routeTarget = options.routeTarget;
     this.routeType = protocol;
     this.match = options.match;
+    this.defaultHostnameRewrite = options.defaultHostnameRewrite;
+    this.pathRewrite = options.pathRewrite;
   }
 
-  public bind(_scope: Construct): GatewayRouteSpecConfig {
-    const prefixPath = this.match ? this.match.prefixPath : '/';
-    if (prefixPath[0] != '/') {
-      throw new Error(`Prefix Path must start with \'/\', got: ${prefixPath}`);
+  public bind(scope: Construct): GatewayRouteSpecConfig {
+    // Set prefix Match to '/' if none on match properties are defined.
+    const prefixMatch = !areMatchPropertiesUndefined(this.match) ? this.match?.path?.bind(scope).requestMatch.prefix :'/';
+    const prefixPathRewrite = this.pathRewrite?.bind(scope).prefix;
+    const pathRewrite = this.pathRewrite?.bind(scope).path;
```

Comment:
Like I said before - do not repeat expressions like `this.pathRewrite?.bind(scope)`.

Reply:
Created the variable `const pathRewriteConfig = this.pathRewrite?.bind(scope);`